cordformation/src/main/kotlin/net/corda/plugins/Dockerform.kt

```diff
@@ -201,6 +207,30 @@ open class Dockerform @Inject constructor(objects: ObjectFactory) : Baseform(obj
             services[it.containerName] = service
         }
 
+        if (external.containerImage.isNotEmpty()) {
+            val extra = mutableMapOf(
+                    "image" to external.containerImage,
+                    "ports" to external.servicePorts.toList().map{QuotedString("${it}:${it}")},
+                    "expose" to external.servicePorts
+            )
+            if(external.volumes.isNotEmpty()){
+                extra["volumes"] = external.volumes.toList().map{ "${it["sourceFile"]}:${it["deploymentPath"]}" }
+            }
+            if(external.environment.isNotEmpty()){
+                extra["environment"] = external.environment.toList().map { "${it.first}=${it.second}" }
+            }
```

Comment:
`environment` is a `MapProperty` and so also has an empty default value.
```kotlin
val environment = external.environment.get()
if (environmen.isNotEmpty()) {
   ...
}
```
Superfluous `toList()` again. This should be enough:
```kotlin
environment.map { "${it.key}=${it.value}" }
```